src/bin/ch-remote.rs

```diff
@@ -3,17 +3,23 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+use std::fmt;
+use std::io::Read;
+use std::marker::PhantomData;
+use std::os::unix::net::UnixStream;
+use std::process;
+
```

Comment:
This whitespace will break the "cargo fmt" ordering (it reorders within groups.) Please make sure you don't add any new whitespace lines in use groups (and remove them if you find them! :-) )

Reply:
I believe that's intentional to follow `StdExternalCrate` strategy while grouping imports, as documented [here](https://rust-lang.github.io/rustfmt/?version=v1.6.0&search=#group_imports).

And I support @rbradford 's view in grouping them together here, if we really need this grouping strategy, we should write it into `rustfmt.toml` and modify our style CI accordingly :)

Reply:
> This whitespace will break the "cargo fmt" ordering (it reorders within groups.) Please make sure you don't add any new whitespace lines in use groups (and remove them if you find them! :-) )

As @TimePrinciple says, the whitespace line is a rustfmt rule, `group_imports="StdExternalCrate"` and it's used to group rust built-in mods, third-party mods and local `self`/`supper`/`crate` mods. This rule is more intuitive and helpful for clarifying the call chain. 

> And I support @rbradford 's view in grouping them together here, if we really need this grouping strategy, we should write it into rustfmt.toml and modify our style CI accordingly :)

I think the fmt CI is the minimum requirement of the code specification, we can do better than CI. We can introduce a stricter CI for rust format in the future if necessary.

Reply:
We do have a ".rustfmt.toml" file that is enforced by the CI. But I don't think there is a way to have it delete the extra whitespace lines.

Our style, that has to be enforced manually (by asking for the removal of empty lines in the code) is that the imports are alphabetically ordered with no groups.

Unfortunately cargo fmt lacks a way to apply no grouping - but I am happy to move to "StdExternalCrate" since that is a style that can be enforced as that will make reviewing easier!.